fix: do not trigger site.changed when draft posts are deleted

im-adithya opened the pull request:
Fixes https://github.com/TryGhost/Ghost/issues/21699

## Description

This adds a check to see if all of the posts being deleted are drafted, and only sets `cacheInvalidate` if not (since `site.changed` is triggered because of `X-Cache-Invalidate` header)

## Checklist

- [x] There's a clear use-case for this code change, explained below
- [x] Commit message has a short title & references relevant issues
- [x] The build will pass (run `yarn test:all` and `yarn lint`)

im-adithya committed:
fix: do not trigger site.changed when draft posts are deleted

im-adithya commented:
@ErisDS can you please have a look :)

ErisDS commented:
Hey there @im-adithya, I see you've made quite a few PRs with bugfixes, really appreciate the effort 🙏 

We have a working practice of always adding a unit test to demonstrate a bug fix, otherwise it's liable to either not be fixed properly, or regress again. How do you feel about adding tests to your changes? 

Cheers!

im-adithya committed:
chore: add test

im-adithya commented:
Hey Hannah! Thanks for noticing, would be more than happy to have them reviewed :)

Absolutely, I added the test now. The test files (atleast the ones in e2e-webhooks) only had the checks for single addition/deletion, so I only checked deletion of a single draft post. Let me know if I should add for bulk deletion too!

im-adithya committed:
chore: typo in test

9larsons committed:
fix: do not trigger site.changed when draft pages are deleted

Applies the draft-aware cache invalidation guard to pages and lightens
the status lookups to fetch only the status column. Adds e2e-webhooks
coverage for the bulk-posts path, draft pages (single + bulk), and
guards that published post/page deletion still invalidates the cache.
